### PR TITLE
Reorganizes first major section of docs

### PR DESCRIPTION
**Pull request by willingc**
It's been a while since we have had a docs sprint. I would like to have one in the future (perhaps at PyCon). In preparation, I would like to get the table of contents for the docs a little more organized and user friendly for newcomers.

This PR reorders the table of contents for the first major section of the docs (Getting Started/Contribution Guide). According to the Sphinx build tool the docs rendered cleanly without error.

Here are a couple of screenshots of the changes to the first section in the docs' Table of Contents:
screen shot 2015-01-11 at 16 31 59

screen shot 2015-01-11 at 16 32 17

### CODE REVIEW

File: docs/getting_started/getting_started.rst [hunk 1]
```diff
@@ -24,24 +24,20 @@ order:
     #. Make a new Github account on https://www.github.com if you don't already have one.
     #. Fork the oh-mainline Github repository located here at https://github.com/openhatch/oh-mainline. Click on the fork button located on the upper  right corner of the project page. Now you have your own personal copy of the oh-mainline repository.
     #. Install git the version control system. If you have already done so, skip to the next step.
-    #. Clone your personal copy of the oh-mainline repository to your computer by typing this command into your terminal::
-
-        $ git clone https://github.com/<YOUR_GITHUB_USERNAME>/oh-mainline.git
+    #. Clone your personal copy of the oh-mainline repository to your computer by typing this command into your terminal
+    
+      git clone https://github.com/<YOUR_GITHUB_USERNAME>/oh-mainline.git
```
Comment (paulproteus) [inline docs/getting_started/getting_started.rst]: FWIW, I like the `$` paradigm that indicates that one is supposed to be typing this at a shell.

File: docs/getting_started/getting_started.rst [hunk 2]
```diff
@@ -12,8 +10,10 @@ good software testing practices. If you have Python experience, you should be
 able to get hacking pretty quickly even if you don't know Django or testing.
 
 
-Getting the source
-==================
+First steps
+===========
+Getting the source code
```
Comment (paulproteus) [inline docs/getting_started/getting_started.rst]: For what it's worth, I suggest a blank line between section headings.

File: docs/getting_started/getting_started.rst [hunk 3]
```diff
@@ -24,24 +24,20 @@ order:
     #. Make a new Github account on https://www.github.com if you don't already have one.
     #. Fork the oh-mainline Github repository located here at https://github.com/openhatch/oh-mainline. Click on the fork button located on the upper  right corner of the project page. Now you have your own personal copy of the oh-mainline repository.
     #. Install git the version control system. If you have already done so, skip to the next step.
-    #. Clone your personal copy of the oh-mainline repository to your computer by typing this command into your terminal::
-
-        $ git clone https://github.com/<YOUR_GITHUB_USERNAME>/oh-mainline.git
+    #. Clone your personal copy of the oh-mainline repository to your computer by typing this command into your terminal
+
```
Comment (paulproteus) [inline docs/getting_started/getting_started.rst]: The Github web view points out that there's unnecessary whitespace at the end of this line.

File: docs/getting_started/getting_started.rst [hunk 4]
```diff
@@ -66,7 +62,7 @@ OpenHatch holds development meetings on IRC; our goal is to hold these meetings
 
 
 Read more documentation
-=======================
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (paulproteus) [inline docs/getting_started/getting_started.rst]: In general I'm +1 on using more section heading types.

One thing I dislike about rST is that it doesn't insist on consistency between documents, which can confuse different people who are editing the same collection of documents, if different docs within the same doc/* directory use different paradigms.

I almost feel like we need an rST style guide, to avoid this kind of internal inconsistency.

File: docs/getting_started/getting_started.rst (hunk 2 above)
Comment (willingc) [inline docs/getting_started/getting_started.rst]: Agreed :)

File: docs/getting_started/getting_started.rst (hunk 3 above)
Comment (willingc) [inline docs/getting_started/getting_started.rst]: Will fix

File: docs/getting_started/getting_started.rst (hunk 1 above)
Comment (willingc) [inline docs/getting_started/getting_started.rst]: Me too. Typo. Will correct.

**Comment by paulproteus** [comment]
I'm very +1 on this as-is. It sounds like you want to add some changes on top before merging it. Feel free to merge this as-is or with minimal changes.

It seems like you're creating a distinction between the "Contributor Guide" (which is a longer doc) and the "Installation guide" (which is the shorter doc). If I understand that correctly, I'm very +1 on that change.

File: docs/getting_started/getting_started.rst (hunk 4 above)
Comment (willingc) [inline docs/getting_started/getting_started.rst]: The Hitchhikers Guide to Python by Kenneth Reitz and team has a good style guide for rST docs. I plan to incorporate with attribution or use it as a guide to an OpenHatch rST style guide.

**Comment by willingc** [comment]
@paulproteus @ehashman @shaunagm @brittag 
I did some reorganizing of the index files for the OpenHatch docs. The screenshots below give a sense of the contents.

I have also added a draft docs style guide which is attributed to Kenneth Reitz's Hitchhiker's Guide to Python.

Sorry for the size of the PR. I would love it if someone could merge. Happy to make any changes to this or in another PR.

screen shot 2015-01-12 at 02 09 11

screen shot 2015-01-12 at 02 09 31

File: docs/index.rst [hunk 5]
```diff
@@ -1,30 +1,142 @@
-====================================
-Welcome to OpenHatch's documentation
-====================================
+====================
+Welcome to OpenHatch
```
Comment (paulproteus) [inline docs/index.rst]: The earlier page said "Welcome to OpenHatch's documentation" because I wonder if people, upon arrival, might think that openhatch.readthedocs.org is the real "OpenHatch website". I'm concerned that people might think this is the "site" not the "docs", if we remove that text.

But maybe this is not a serious concern, or maybe it's one that your text changes below address. I'm curious what you think.

I'm -0 to shipping this as-is with that possible source of confusion, but I can live with it.

File: docs/index.rst [hunk 6]
```diff
@@ -1,30 +1,142 @@
...
+    internals/index
     tutorials/index
     community/index
-    internals/index
-    CONTRIBUTING.rst
-    README.rst
 
-:ref:`search`
 
+Contributors Guide
```
Comment (paulproteus) [inline docs/index.rst]: Probably this should be one of:

* Contributors' Guide
* Contributor's Guide
* Contribution Guide

Python.org goes with the middle one: https://docs.python.org/devguide/ "Developer's Guide"

File: docs/index.rst [hunk 7]
```diff
@@ -1,30 +1,142 @@
...
+
+   getting_started/project_overview
+   getting_started/getting_started
+   getting_started/installation
+   getting_started/testing_basics
+   getting_started/documentation
+   getting_started/handling_contributions
+
+
+Technical Guide to Development and Documentation
```
Comment (paulproteus) [inline docs/index.rst]: +1 I like this section heading a lot.

File: docs/getting_started/getting_started.rst (hunk 4 above)
Comment (paulproteus) [inline docs/getting_started/getting_started.rst]: +1

My preference would be to link to it, rather than include a copy.

File: docs/advanced/index.rst [hunk 8]
```diff
@@ -1,17 +1,18 @@
...
+   :maxdepth: 2
 
-    advanced_installation
-    layout_source_code
-    deployment
-    heroku
-    maintenance
-    adding_new_bug_tracker_git
-    advanced_testing
+   styleguide
```
Comment (paulproteus) [inline docs/advanced/index.rst]: When I build this locally, I guet:

/home/paulproteus/projects/oh-mainline/docs/index.rst:56: WARNING: toctree contains reference to nonexisting document u'advanced/styleguide'

Is this the rST styleguide you mentioned? I think you didn't `git add` it.

FWIW, if possible, I would prefer to include that by reference, rather than including a copy, but I can live with it either way.

**Comment by paulproteus** [comment]
I'm +1 on merging this as-is; any further responses to the above can be separate pull requests, I figure.

So I'll just merge it right now!

**Comment by willingc** [comment]
@paulproteus I agree with your comments. I will submit a separate PR for the comments that you made and for the left side bar rendering on rtd which differs from Sphinx. Thanks!

**Comment by paulproteus** [comment]
Yippee!

File: docs/index.rst (hunk 6 above)
Comment (ehashman) [inline docs/index.rst]: +1 option 1 or 2, -1 option 3

**Comment by ehashman** [comment]
PS @willingc this is awesome :D
